src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/common/stateful_init_graph.cpp

```diff
@@ -273,6 +287,134 @@ TEST_P(InitGraphStatefulDiffPrimitiveModel, CompareWithRefs) {
+            create_constant_node(netPrc, mul_shape, gen));
+
+        // FakeConvert_1
+        const ov::Shape scale_shape = {1}, shift_shape = {1};
+        const auto fake_convert_1 = std::make_shared<ov::op::v13::FakeConvert>(mul_0,
+            create_constant_node(netPrc, scale_shape, gen),
+            create_constant_node(netPrc, shift_shape, gen),
+            ov::element::f8e4m3);
+
+        // Convert_1
+        const ov::Shape convert_shape = {targetStaticShapes[0][0][-1], 1};
+        auto convert_1 = std::make_shared<ov::op::v0::Convert>(
+            create_constant_node(ov::element::f8e4m3, convert_shape, gen), netPrc);
+
+        // Multiply_1
+        const auto mul_1 = std::make_shared<ov::op::v1::Multiply>(convert_1,
+            create_constant_node(netPrc, mul_shape, gen));
+
+        // MatMul_1
+        auto matmul_1 = std::make_shared<ov::op::v0::MatMul>(fake_convert_1, mul_1);
+
+        // ReadValue
+        statePrc = ov::element::f32;
+        auto variable = std::make_shared<ov::op::util::Variable>(
+            ov::op::util::VariableInfo{inputDynamicShapes[0], statePrc, "var"});
+        auto readvalue = std::make_shared<ov::op::v6::ReadValue>(matmul_1, variable);
+
+        // FakeConvert_2
+        std::shared_ptr<ov::Node> fake_convert_2 = std::make_shared<ov::op::v13::FakeConvert>(readvalue,
+            create_constant_node(netPrc, scale_shape, gen),
+            create_constant_node(netPrc, shift_shape, gen),
+            ov::element::f8e4m3);
+
+        // Assign
+        auto assign = std::make_shared<ov::op::v6::Assign>(directPair ? readvalue : fake_convert_2, variable);
+
+        // Convert_2
+        auto convert_2 = std::make_shared<ov::op::v0::Convert>(
+            create_constant_node(ov::element::f8e4m3, convert_shape, gen), netPrc);
+
+        // Multiply_2
+        const auto mul_2 = std::make_shared<ov::op::v1::Multiply>(convert_2,
+            create_constant_node(netPrc, mul_shape, gen));
+
+        // MatMul_2
+        auto matmul_2 = std::make_shared<ov::op::v0::MatMul>(fake_convert_2, mul_2);
+
+        // Convert_3
+        auto convert_3 = std::make_shared<ov::op::v0::Convert>(
+            create_constant_node(ov::element::f8e4m3, convert_shape, gen), netPrc);
+
+        // Multiply_3
+        const auto mul_3 = std::make_shared<ov::op::v1::Multiply>(convert_3,
+            create_constant_node(netPrc, mul_shape, gen));
+
+        // MatMul_3
+        auto matmul_3 = std::make_shared<ov::op::v0::MatMul>(fake_convert_1, mul_3);
+
+        // Add
+        auto add = std::make_shared<ov::op::v1::Add>(matmul_2, matmul_3);
+
+        // Result
+        auto result = std::make_shared<ov::op::v0::Result>(add);
+
+        function = std::make_shared<ov::Model>(ov::ResultVector({result}), ov::SinkVector({assign}), input_params);
+    }
+
+    void check_init_graph_node() override {
+        CheckNumberOfNodesWithType(compiledModel, "FakeConvert", 0);
```

Comment:
Shouldn't we check the other nodes count to make sure that the init subgraphs are indeed rolled into the MemoryInput nodes?

Reply:
Good point! Applied. I've added the key node FullyConnected in monitoring. As there are 3 FullyConnected in total, 1 should be rolled into MemoryInput-subgraph, 2 should be left.